lib/phobos_db_checkpoint/handler.rb

```diff
@@ -31,7 +45,15 @@ def around_consume(payload, metadata)
           end
 
           event_action = instrument('db_checkpoint.event_action', event_metadata) do
-            yield
+            begin
```

Comment:
Exactly, we can't take for granted that a message can be skipped and the next will succeed. It can have requirements on the previous message, it's up to the developer to configure phobos to fits its needs.